### PR TITLE
Feature/favorites

### PR DESCRIPTION
**Pull request by ProgrammingAStorm**
# Issue #4 

# Complete

- User can press the heart icon on a pet's card to add it to the favorites.
- Favorites are saved in localStorage.
- Favorites are loaded from localStorage and are updated every time the heart icon is clicked.

# Remaing To Do

- Function that populates a modal with the queried data.

### CODE REVIEW

File: assets/js/script.js [hunk 1]
```diff
@@ -11,15 +76,13 @@ const getToken = () => {
         data: `grant_type=client_credentials&client_id=${ID}&client_secret=${SECRET}`,
         // Upon Success, stores token with a key of 'token'.
         success: function(data) {
-            getAnimals(data);
```
Comment (cmskorcz) [inline assets/js/script.js:14]: `getAnimals(data)` needs to be kept, and the localStorage needs to be removed

File: assets/js/script.js [hunk 2]
```diff
@@ -1,8 +1,73 @@
-const ID = '5ojBPITunTAqqOTxJtnD0gylUmUo3CCjakTjPZsivTFN5D9wc4'
-const SECRET = 'j0E7b0TKzxU0qUmPNSGZJ7dsfKpEkgW7PKhr4bvt'
-
+const ID = '5ojBPITunTAqqOTxJtnD0gylUmUo3CCjakTjPZsivTFN5D9wc4';
+const SECRET = 'j0E7b0TKzxU0qUmPNSGZJ7dsfKpEkgW7PKhr4bvt';
```
Comment (cmskorcz) [inline assets/js/script.js:2]: Initialize TOKEN with `let TOKEN`

File: assets/js/script.js (hunk 1 above)
Comment (cmskorcz) [reply assets/js/script.js:14]: Actually just remove `getAnimals(data)` from this line

File: assets/js/script.js [hunk 3]
```diff
@@ -11,15 +76,13 @@ const getToken = () => {
         data: `grant_type=client_credentials&client_id=${ID}&client_secret=${SECRET}`,
         // Upon Success, stores token with a key of 'token'.
         success: function(data) {
-            getAnimals(data);
+            localStorage.setItem('token', data.access_token);
```
Comment (cmskorcz) [inline assets/js/script.js:79]: Replace with `TOKEN = data.access_token;
            setTimeout(() => {
                TOKEN = ''
            }, (3600 * 1000))
`

File: assets/js/script.js [hunk 4]
```diff
@@ -11,15 +76,13 @@ const getToken = () => {
...
         },
         dataType: "json"
-      })
+    })
 };
-
-// // Uses token to access array of animals, will need to add location query eventually
+// Uses token to access array of animals, will need to add location query eventually
 const getAnimals = (data) => {
```
Comment (cmskorcz) [inline assets/js/script.js:85]: In `getAnimals()` delete `(data)` parameter, and change `data.access_token` to `TOKEN`

File: assets/js/script.js [hunk 5]
```diff
@@ -37,4 +100,44 @@ const getAnimals = (data) => {
     })
 };
 
-$tokenButton.on('click', getToken);
\ No newline at end of file
+$tokenButton.on('click', getToken);
```
Comment (cmskorcz) [inline assets/js/script.js:103]: `getToken` needs to be changed to `getAnimals`

File: assets/js/script.js [hunk 6]
```diff
@@ -37,4 +100,44 @@ const getAnimals = (data) => {
...
+        $petContent.children(".name").text(),
+        $petContent.attr("data-id")
+    );
+});
+
+//When a favorites button is pressed it uses the attached ID to query the specifc animal.
+$("#favorites").on("click", "button", function() {
+    getToken()
+
+    let TOKEN = localStorage.getItem('token');
```
Comment (cmskorcz) [inline assets/js/script.js:122]: remove this line

File: assets/js/script.js [hunk 7]
```diff
@@ -37,4 +100,44 @@ const getAnimals = (data) => {
...
+        success: function(data) {
+            /*****************************TO DO*********************************
+            *On success the data needs to be used to open and populate a modal.*
+            *******************************************************************/
+            console.log(data)
+        },
+        dataType: "json"
+    })
+});
+            getAnimals(data);
```
Comment (cmskorcz) [inline assets/js/script.js:139]: remove data as an argument

File: assets/js/script.js [hunk 8]
```diff
@@ -37,4 +100,44 @@ const getAnimals = (data) => {
...
+        },
+        dataType: "json"
+      })
+};
```
Comment (cmskorcz) [inline assets/js/script.js:143]: at bottom of the script add in `getToken()` and `setInterval(() => { getToken() }, (3600 * 1000));`

**Review by cmskorcz — CHANGES_REQUESTED** [review]
These are the changes to be made to avoid merge conflicts

**Review by ProgrammingAStorm — COMMENTED** [review]
All done

**Review by cmskorcz — APPROVED** [review]
lgtm

**Pull request merged**